client/src/cmdlfem4x50.c

```diff
@@ -246,270 +121,501 @@ static void print_info_result(uint8_t *data, bool verbose) {
     PrintAndLogEx(NORMAL, "");
 }
 
-//quick test for EM4x50 tag
-bool detect_4x50_block(void) {
-    em4x50_data_t etd = {
-        .pwd_given = false,
-        .addr_given = true,
-        .address = EM4X50_DEVICE_ID,
-    };
-    em4x50_word_t words[EM4X50_NO_WORDS];
-    return (em4x50_read(&etd, words, false) == PM3_SUCCESS);
+//==============================================================================
+// file/memory functions
+//==============================================================================
+
+static int em4x50_load_file(const char *filename, uint8_t *data, size_t data_len, size_t *bytes_read) {
+
+    // read data from dump file; file type is derived from file name extension
+
+    int res = 0;
+    uint32_t serial = 0x0, device_id = 0x0;
+     
+    if (str_endswith(filename, ".eml"))
+        res = loadFileEML(filename, data, bytes_read) != PM3_SUCCESS;
+    else if (str_endswith(filename, ".json"))
+        res = loadFileJSON(filename, data, data_len, bytes_read, NULL);
+    else
+        res = loadFile(filename, ".bin", data, data_len, bytes_read);
+
+    if ((res != PM3_SUCCESS) && (*bytes_read != DUMP_FILESIZE))
+        return PM3_EFILE;
+
+    // valid em4x50 data?
+    serial = bytes_to_num(data + 4 * EM4X50_DEVICE_SERIAL, 4);
+    device_id = bytes_to_num(data + 4 * EM4X50_DEVICE_ID, 4);
+    if (serial == device_id) {
+        PrintAndLogEx(WARNING, "No valid em4x50 data in file %s.", filename);
+        return PM3_ENODATA;
+    }
+
+    return PM3_SUCCESS;
 }
 
-int read_em4x50_uid(void) {
-    em4x50_data_t etd = {
-        .pwd_given = false,
-        .addr_given = true,
-        .address = EM4X50_DEVICE_SERIAL,
-    };
-    em4x50_word_t words[EM4X50_NO_WORDS];
-    int res = em4x50_read(&etd, words, false);
-    if (res == PM3_SUCCESS)
-        PrintAndLogEx(INFO, " Serial: " _GREEN_("%s"), sprint_hex(words[EM4X50_DEVICE_SERIAL].byte, 4));
-    return res;
+static int em4x50_wipe_flash(int page) {
+    
+    int isok = 0;
+    
+    clearCommandBuffer();
+    SendCommandMIX(CMD_FLASHMEM_WIPE, page, false, 0, NULL, 0);
```

Comment:
flashmem_wipe??  ..
We use SPIFFS if you wanna save files.  
And this needs to be guarded with checks for RDV4 w flashmem enabled..